accelerator/mps_accelerator.py

```diff
@@ -0,0 +1,230 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+import os
+import pkgutil
+import importlib
+
+from .abstract_accelerator import DeepSpeedAccelerator
+
+# During setup stage torch may not be installed, pass on no torch will
+# allow op builder related API to be executed.
+try:
+    import torch.mps
+except ImportError:
+    pass
+
+
+class MPS_Accelerator(DeepSpeedAccelerator):
+    def __init__(self):
+        self._name = "mps"
+        self._communication_backend_name = None
+
+    def is_synchronized_device(self):
+        return False
+
+    # Device APIs
+
+    def device_name(self, device_index):
+        return "mps"
```

Comment:
I ran your branch on my M1 macbook air. DeepSpeed is failing to install because there's no default value for `device_index` here. I did get it to successfully install If I changed

You can see an example here: https://github.com/microsoft/DeepSpeed/blob/master/accelerator/cuda_accelerator.py#L29-L32

I did get it to successfully install if i changed this to:

```python
    def device_name(self, device_index=None):
        if device_index == None:
            return 'mps'
        return 'mps:{}'.format(device_index)
```

It appears mps allows you to pass a device index in. If you set it to non-zero it seems to automatically replace it with 0 though :)

Reply:
Thank you soo much for your feedback. I have changed the code accordingly